proposed/SerializerGoals5.0.md

```diff
@@ -0,0 +1,281 @@
+        {
+            var p = new JsonPropertyInfo<DateTime>();
+            p.Options = Options;
+            p.Converter = Serializer.DateTimeConverter;
+            p.NameAsString = nameof(WeatherForecast.Date);
+
+            var properties = new List<JsonPropertyInfo>(4);
+
+            properties.Add(new JsonPropertyInfo<DateTime>
+            {
+                Options = Options,
+                Converter = Serializer.DateTimeConverter,
+                NameAsString = nameof(WeatherForecast.Date),
+                HasGetter = true,
+                HasSetter = true,
+                ShouldSerialize = true,
+                ShouldDeserialize = true,
+                // These delegates are nice that they work with aggressive linker
+                // but slower due to the extra hop (which may be avoided with
+                // new reflection features).
+                Get = (obj) => { return ((WeatherForecast)obj).Date; },
+                Set = (obj, value) => { ((WeatherForecast)obj).Date = value; }
+            });
+
+            properties.Add(new JsonPropertyInfo<int>
+            {
+                Options = Options,
+                Converter = Serializer.Int32Converter,
+                NameAsString = nameof(WeatherForecast.TemperatureC),
+                HasGetter = true,
+                HasSetter = true,
+                Get = (obj) => { return ((WeatherForecast)obj).TemperatureC; },
+                Set = (obj, value) => { ((WeatherForecast)obj).TemperatureC = value; }
+            });
+
+            properties.Add(new JsonPropertyInfo<int>
+            {
+                Options = Options,
+                Converter = Serializer.Int32Converter,
+                HasGetter = true,
+                NameAsString = nameof(WeatherForecast.TemperatureF),
+                Get = (obj) => { return ((WeatherForecast)obj).TemperatureF; },
+            });
+
+            properties.Add(new JsonPropertyInfo<string>
+            {
+                Options = Options,
+                Converter = Serializer.StringConverter,
+                NameAsString = nameof(WeatherForecast.Summary),
+                HasGetter = true,
+                HasSetter = true,
+                Get = (obj) => { return ((WeatherForecast)obj).Summary; },
+                Set = (obj, value) => { ((WeatherForecast)obj).Summary = value; }
+            });
+
+            return properties;
+        }
+    }
+}
+```
+</details>
+
+## Object converter + code-gen
+This is simular to "Metadata provider code-gen" but includes exposing a new type of "object converter". This type of converter, however, exposes the raw reader\writer and has performance issues with Streams due to requiring read-ahead.
+
+# Code-Gen
+For the options that use code-gen there are several issues to discuss; this is covered in other documention but one open issue is to determine what Types to add code-gen for. The two basic approaches are automatic and explicit:
```

Comment:
Are there links for the other documentation?

Reply:
I'll add them.

Per @layomia see [publicly available markdown with design notes/questions/benchmarks](https://gist.github.com/layomia/5221e73374fd74dc360e74731dacfdb5) and [code-gen prototype](https://github.com/layomia/jsonconvertergenerator).